modules/sfm/CMakeLists.txt

```diff
@@ -113,7 +113,7 @@ if(NOT CMAKE_VERSION VERSION_LESS 2.8.11) # See ocv_target_include_directories()
   endif()
 endif()
 include_directories(${OCV_TARGET_INCLUDE_DIRS_${the_module}})
-add_subdirectory(src/libmv_light)
+add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/src/libmv_light)
```

Comment:
Try this:
```
add_subdirectory("${CMAKE_CURRENT_LIST_DIR}/src/libmv_light" "${CMAKE_CURRENT_BINARY_DIR}/libmv")
```

/cc @LaurentBerger

Reply:
--     module opencv_optflow...
--     module opencv_sfm...
-- No preference for use of exported gflags CMake configuration set, and no hints for include/library directories provided. Defaulting to preferring an installed/exported gflags CMake configuration if available.
-- Found installed version of gflags: G:/lib/build/gflags
-- Detected gflags version: 2.2.1
-- Found required Ceres dependency: Eigen version 3.3.90 in G:/Lib/eigen
-- Found installed version of glog: G:/lib/Install/glog/lib/cmake/glog
-- Detected glog version: 0.3.5
-- Found required Ceres dependency: glog
-- Found Ceres version: 1.13.0 installed in: G:/Lib/install/ceres-solver with components: [SchurSpecializations, OpenMP]
-- Checking SFM deps... TRUE
--     module opencv_stitching...
-- Processing WORLD modules... DONE
-- Excluding from source files list: G:/Lib/build/opencv/modules/world/accum.neon.cpp
-- Processing module opencv_img_hash...

👍 

Ok but some compilation errors


in multiview project _USE_MATH_DEFINES is not defined in multiwiew project (libmv_light) .

In project correspondance errors are :  
1>G:\Lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\correspondence\feature_matching.cc(21): fatal error C1083: Impossible d'ouvrir le fichier include : 'opencv2/features2d.hpp' : No such file or directory
1>G:\Lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\correspondence\nRobustViewMatching.cc(21): fatal error C1083:cannot open include : 'opencv2/highgui.hpp' : No such file or directory
1>G:\Lib\opencv_contrib\modules\sfm\src\libmv_light\libmv/correspondence/matches.h(27): fatal error C1083: cannot open include : 'opencv2/core.hpp' : No such file or directory (compilation du fichier source G:\Lib\opencv_contrib\modules\sfm\src\libmv_light\libmv\correspondence\matches.cc)



in this project include are (opencv sources are in G:\Lib\opencv and build project in G:\Lib\build\opencv)
G:\Lib\opencv\include\opencv;G:\Lib\opencv_contrib\modules\sfm\src\libmv_light;G:\Lib\build\opencv\modules\world;G:\Lib\opencv\3rdparty\libjasper;G:\Lib\build\opencv\3rdparty\libtiff;G:\Lib\opencv\3rdparty\libtiff;G:\Lib\opencv\3rdparty\libpng;G:\Lib\opencv\3rdparty\libwebp;G:\Lib\opencv\3rdparty\libjpeg;G:\Lib\build\opencv\3rdparty\zlib;G:\Lib\opencv\3rdparty\zlib;G:\Lib\build\opencv\3rdparty\ippicv\ippicv_win\include;G:\Lib\build\opencv\3rdparty\ippicv\ippicv_win\..\ippiw_win\include;G:\Lib\build\opencv;G:\Lib\eigen;G:\Lib\Install\vtk\include\vtk-8.1;G:\Lib\opencv\3rdparty\openexr\Half;G:\Lib\opencv\3rdparty\openexr\Iex;G:\Lib\opencv\3rdparty\openexr\IlmThread;G:\Lib\opencv\3rdparty\openexr\Imath;G:\Lib\opencv\3rdparty\openexr\IlmImf;G:\Lib\opencv\3rdparty\include;G:\Lib\opencv\3rdparty\include\ffmpeg_;G:\lib\Install\tesseract\include;G:\Lib\build\gflags\include;G:\Lib\install\ceres-solver\include;G:\Lib\install\glog\include;%(AdditionalIncludeDirectories)